module/VuFind/src/VuFind/Controller/LibraryCardsController.php

```diff
@@ -240,6 +245,42 @@ public function selectCardAction()
         return $this->redirect()->toRoute('myresearch-home');
     }
 
+    /**
+     * Redirects to Shibboleth authentication to connect a new library card
+     *
+     * @return \Laminas\Http\Response
+     */
+    public function connectNewShibbolethCardAction()
+    {
+        $url = $this->getServerUrl('librarycards-connectshibbolethcard');
+        $redirectUrl = $this->getAuthManager()->getSessionInitiator($url);
+        if ($redirectUrl == null) {
+            $this->flashMessenger()
+                ->addMessage('authentication_error_technical', 'error');
+            $redirectUrl = '/LibraryCards/Home';
```

Comment:
It would be better to return a redirect with `->toRoute()` to avoid hard-coding paths.